### PR TITLE
New FILA Report Layout + Generation by date range

### LINKED ISSUE #251: Allow the user to visualise the FILA according to a selected range of date

### LINKED ISSUE #252: Adjust fila report according to the standard format

### PR DESCRIPTION
**Pull request by edrisse**
Fix #251 
Fix #252

### CODE REVIEW

**Comment by edrisse** [comment]
[Bug: FILA report must be according to layout and must be generated by date range.](https://trello.com/c/FWPV1NIu/246-bug-fila-report-must-be-according-to-layout-and-must-be-generated-by-date-range)

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 1]
```diff
@@ -3,97 +3,111 @@
...
+        <tr>
+          <td colspan="4" style="border: none; text-align: left;">* O farmacêutico não pode aviar mais MARVs depois da expiração da receita médica</td>
+        </tr>
+      </tfoot>
+    </table>
+
+    <table style="width: 144mm; margin-left: 1mm;">
+      <thead>
+      <tr>
+        <th>Data de Levantamento de ARVs</th>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Missing translation.

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 2]
```diff
@@ -3,97 +3,111 @@
...
+        <th class="medicine-column">Medicamentos ARVs</th>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Missing translation.

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 3]
```diff
@@ -3,97 +3,111 @@
...
+        <th>Quantidade Aviada</th>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Missing translation.

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 4]
```diff
@@ -3,97 +3,111 @@
...
+        <th class="dosage-column">Dosagem</th>
+        <th>Data do Próximo Levantamento</th>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Missing translation.

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 5]
```diff
@@ -3,97 +3,111 @@
...
+          <th>Data do Próximo Levantamento</th>
+        </tr>
+      </thead>
+      <tbody >
+        <tr ng-repeat="displacement in filaDisplacements1 track by $index" class="displacement-row">
+          <td ng-if="displacement.dateRowSpanSize > 0" rowspan="{{displacement.dateRowSpanSize}}">{{displacement.date | date:"dd/MM/yyyy"}}&nbsp;</td>
+          <td>{{displacement.medicine}}&nbsp;</td>
+          <td>{{displacement.quantity}}&nbsp;</td>
+          <td>{{displacement.dosage}}&nbsp;</td>
+          <td>{{displacement.nextDisplacement | date:"dd/MM/yyyy"}}&nbsp;</td>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Date format should not be hard-coded. For this case ```{{displacement.nextDisplacement | date}}``` is sufficient. See: https://github.com/esaude/esaude-emr-poc/blob/master/app/common/application/constants.js and ```dateFilterDecorator``` on https://github.com/esaude/esaude-emr-poc/blob/master/app/common/application/config.js.

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 6]
```diff
@@ -3,97 +3,111 @@
...
+        <th>Data do Próximo Levantamento</th>
+      </tr>
+      </thead>
+      <tbody >
+      <tr ng-repeat="displacement in filaDisplacements2 track by $index" class="displacement-row">
+        <td ng-if="displacement.dateRowSpanSize > 0" rowspan="{{displacement.dateRowSpanSize}}">{{displacement.date | date:"dd/MM/yyyy"}}&nbsp;</td>
+        <td>{{displacement.medicine}}&nbsp;</td>
+        <td>{{displacement.quantity}}&nbsp;</td>
+        <td>{{displacement.dosage}}&nbsp;</td>
+        <td>{{displacement.nextDisplacement | date:"dd/MM/yyyy"}}&nbsp;</td>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Date format should not be hard-coded.

File: app/pharmacy/views/fila-history.html [hunk 7]
```diff
@@ -1,54 +1,69 @@
...
-    <tr ng-repeat="p in vm.displayedPickups">
-      <td>{{p.encounterDatetime | date}}</td>
-      <td>{{p.regimen}}</td>
-      <td>{{p.posology}}</td>
-      <td>{{p.quantity}}</td>
-      <td>{{p.nextPickup | date}}</td>
-      <td>{{p.provider}}</td>
-    </tr>
+      <tr ng-repeat="p in vm.displayedPickups">
+        <td>{{p.dispensationItemCreationDate | date:"dd/MM/yyyy"}}</td>
```
Comment (ynurmahomed) [inline app/pharmacy/views/fila-history.html]: Date format should not be hard-coded.

File: app/pharmacy/views/fila-history.html [hunk 8]
```diff
@@ -1,54 +1,69 @@
...
+        <td>{{p.drugOrder.drug.display}}</td>
+        <td>{{p.drugOrder.dose}} {{p.drugOrder.doseUnits.display}} {{p.drugOrder.frequency.display}}</td>
+        <td>{{p.quantityDispensed}}</td>
+        <td>{{p.dateOfNextPickUp | date:"dd/MM/yyyy"}}</td>
```
Comment (ynurmahomed) [inline app/pharmacy/views/fila-history.html]: Date format should not be hard-coded.

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 9]
```diff
@@ -3,97 +3,111 @@
...
+          <th>{{'DISPENSATION_DATE' | translate}}</th>
+          <th class="medicine-column">Medicamentos ARVs</th>
+          <th>Quantidade Aviada</th>
+          <th class="dosage-column">Dosagem</th>
+          <th>Data do Próximo Levantamento</th>
+        </tr>
+      </thead>
+      <tbody >
+        <tr ng-repeat="displacement in filaDisplacements1 track by $index" class="displacement-row">
+          <td ng-if="displacement.dateRowSpanSize > 0" rowspan="{{displacement.dateRowSpanSize}}">{{displacement.date | date:"dd/MM/yyyy"}}&nbsp;</td>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Date format should not be hard-coded.

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 10]
```diff
@@ -3,97 +3,111 @@
...
+    <h2 id="top-right-area-title">{{'FILA_REPORT_PERSONAL_DATA_NOTE' | translate}}</h2>
+    <div id="patient-data-container">
+      <div style="font-size: 8pt; font-weight: bold; width: 40mm; margin-left: 3mm; align-self: center; font-size: 10pt; text-align: center;">
+        <span id="nid-label">{{'SEARCH_PATIENT_NID' | translate}}:</span> {{patient.identifier}}
+      </div>
+      <div id="personal-data-conteiner">
+        <div id="first-line-personal-data" class="personal-data">Nº do Livro TARV: ______, Pág: _____, Linha: _____</div>
+        <div class="personal-data">{{'SEARCH_PATIENT_FIRSTNAME' | translate}}: {{patient.fullName}}</div>
+        <div class="personal-data">{{'PATIENT_INFO_OTHERS' | translate}}: {{patient['Numero de Telefone 1']}} / {{patient['Numero de Telefone 2']}}</div>
+        <div class="personal-data">{{'PATIENT_INFO_ADDRESS' | translate}}: {{patient.address.display}}</div>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Address is not showing.

image

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 11]
```diff
@@ -3,97 +3,111 @@
...
-          </tbody>
-        </table>
+  <div id="top-right-area">
+    <h2 id="top-right-area-title">{{'FILA_REPORT_PERSONAL_DATA_NOTE' | translate}}</h2>
+    <div id="patient-data-container">
+      <div style="font-size: 8pt; font-weight: bold; width: 40mm; margin-left: 3mm; align-self: center; font-size: 10pt; text-align: center;">
+        <span id="nid-label">{{'SEARCH_PATIENT_NID' | translate}}:</span> {{patient.identifier}}
+      </div>
+      <div id="personal-data-conteiner">
+        <div id="first-line-personal-data" class="personal-data">Nº do Livro TARV: ______, Pág: _____, Linha: _____</div>
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Translation missing.

File: app/patient-details/views/patient-arv-pickup-history-report.html [hunk 12]
```diff
@@ -3,97 +3,111 @@
 <link href="../../styles/patient-arv-pickup-history-report.css" rel="stylesheet" media="print">
 <!-- endbuild -->
 
-<div class="arv-pickup container-fluid">
+<div class="header-container" style="display: flex;">
```
Comment (ynurmahomed) [inline app/patient-details/views/patient-arv-pickup-history-report.html]: Preferable we should use stylesheets instead of inline styles.

**Pull request merged**